libs/ledger/src/chain/block_coordinator.cpp

```diff
@@ -1274,6 +1312,7 @@ void BlockCoordinator::Reset()
   last_executed_block_.ApplyVoid([](auto &digest) { digest = chain::ZERO_HASH; });
   execution_manager_.SetLastProcessedBlock(chain::ZERO_HASH);
   chain_.Reset();
+  current_block_ = chain_.GetHeaviestBlock();
```

Comment:
We should make heavier use of this reset function when there is an extinction level event